lab_1/main.py

```diff
@@ -2,45 +2,110 @@
-    """
-    Removes stop words
-    :param tokens: a list of tokens
-    :param stop_words: a list of stop words
-    :return: a list of tokens without stop words
-    """
-    pass
+
+    filt_tokens = []
+    for token in tokens:
+        if token not in stop_words:
+            filt_tokens.append(token)
+            tokens = filt_tokens
+    return tokens
+
+
+tokens_en = tokenize(en_text)
+tokens_de = tokenize(de_text)
+tokens_unknown = tokenize(unknown_text)
+
+stop_words_en = nltk.corpus.stopwords.words('english')
+stop_words_de = nltk.corpus.stopwords.words('german')
+stop_words_unknown = []
+
+
+# print(remove_stop_words(tokens_en, stop_words_en))
+# print(remove_stop_words(tokens_de, stop_words_de))
+# print(remove_stop_words(tokens_unknown, stop_words_unknown))
 
 
 def calculate_frequencies(tokens: list) -> dict or None:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens
-    :return: a dictionary with frequencies
-    """
-    pass
+
+    freq_dict = {}
+    for word in tokens:
+        if word not in freq_dict:
+            freq_dict[word] = 1
+        else:
+            freq_dict[word] += 1
+    return freq_dict
+
+
+tokens_en = remove_stop_words(tokens_en, stop_words_en)
+tokens_de = remove_stop_words(tokens_de, stop_words_de)
+tokens_unknown = remove_stop_words(tokens_unknown, stop_words_unknown)
 
 
+# print(calculate_frequencies(tokens_en))
+# print(calculate_frequencies(tokens_de))
+# print(calculate_frequencies(tokens_unknown))
+
 def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
-    """
-    Returns the most common words
-    :param freq_dict: a dictionary with frequencies
-    :param top_n: a number of the most common words
-    :return: a list of the most common words
-    """
-    pass
+
+    words = len(freq_dict)
+    freq_val = reversed(sorted(freq_dict.values()))
+    sorted_freq = {}
+
+    for i in freq_val:
+        for k in freq_dict.keys():
+            if freq_dict[k] == i:
+                sorted_freq[k] = freq_dict[k]
+
+    sf_val = [sorted_freq.values()]
+
+    if top_n > words:
+        top_words = sorted_freq
+        return top_words
+    if top_n < words:
+        s = list(sorted_freq.keys())[:(top_n)]
+        v = list(sorted_freq.values())
+        top_words = dict(zip(s, v))
+        return top_words
+
+
+freq_dict_en = calculate_frequencies(tokens_en)
+freq_dict_de = calculate_frequencies(tokens_de)
+freq_dict_unknown = calculate_frequencies(tokens_unknown)
+
+new_dicts_en = {v: k for k, v in freq_dict_en.items()}
+new_dicts_de = {v: k for k, v in freq_dict_de.items()}
+new_dicts_unknown = {v: k for k, v in freq_dict_unknown.items()}
+
+top_n_en = len(new_dicts_en)
+top_n_de = len(new_dicts_de)
+top_n_unknown = len(new_dicts_unknown)
+
+
+# print(get_top_n_words(freq_dict_en, top_n_en))
+# print(get_top_n_words(freq_dict_de, top_n_de))
+# print(get_top_n_words(freq_dict_unknown, top_n_unknown))
```

Comment:
The `main.py` file should not contain function calls - you should check your implementation using the `start.py` file.